src/CommunityToolkit.Maui.MediaElement/Views/MauiMediaElement.windows.cs

```diff
@@ -20,9 +43,158 @@ public class MauiMediaElement : Grid, IDisposable
+		Children.Add(ButtonContainer);
+
+		mediaPlayerElement.PointerMoved += MediaPlayerElement_PointerMoved;
+	}
+
+	void MediaPlayerElement_PointerMoved(object sender, PointerRoutedEventArgs e)
+	{
+		e.Handled = true;
+		mediaPlayerElement.PointerMoved -= MediaPlayerElement_PointerMoved;
+		InitializeTimer();
+		ButtonContainer.Visibility = Microsoft.UI.Xaml.Visibility.Visible;
+	}
+
+	void InitializeTimer()
+	{
+		if (timer is not null)
+		{
+			return;
+		}
+
+		timer = new();
+		timer.Tick += OnTimer_Tick;
+		timer.Interval = TimeSpan.FromSeconds(5);
+		timer.Start();
+	}
+
+	void OnTimer_Tick(object? sender, object e)
+	{
+		ClearTimer();
+		ButtonContainer.Visibility = Microsoft.UI.Xaml.Visibility.Collapsed;
+		mediaPlayerElement.PointerMoved += MediaPlayerElement_PointerMoved;
+	}
+
+	void ClearTimer()
+	{
+		if (timer is null)
+		{
+			return;
+		}
+
+		timer.Tick -= OnTimer_Tick;
+		timer.Stop();
+		timer = null;
 	}
 
+	void ButtonSetFullScreenStatus(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
+	{
+		var currentPage = CurrentPage;
+		if (appWindow is null)
+		{
+			return;
+		}
+
+		if (appWindow.Presenter.Kind == AppWindowPresenterKind.FullScreen)
+		{
+			appWindow.SetPresenter(AppWindowPresenterKind.Default);
+			Shell.SetNavBarIsVisible(CurrentPage, navbarExist);
+
+			if (popup.IsOpen)
+			{
+				popup.IsOpen = false;
+				popup.Child = null;
+				grid.Children.Clear();
+			}
+
+			Children.Add(this.mediaPlayerElement);
+			Children.Add(this.ButtonContainer);
+
+			var parent = mediaPlayerElement.Parent as FrameworkElement;
+			mediaPlayerElement.Width = parent?.Width ?? mediaPlayerElement.Width;
+			mediaPlayerElement.Height = parent?.Height ?? mediaPlayerElement.Height;
+		}
+		else
+		{
+			appWindow.SetPresenter(AppWindowPresenterKind.FullScreen);
+			navbarExist = Shell.GetNavBarIsVisible(currentPage);
+			Shell.SetNavBarIsVisible(CurrentPage, false);
+
+			var displayInfo = DeviceDisplay.Current.MainDisplayInfo;
+			mediaPlayerElement.Width = displayInfo.Width / displayInfo.Density;
+			mediaPlayerElement.Height = displayInfo.Height / displayInfo.Density;
+
+			Children.Clear();
+			grid.Children.Add(mediaPlayerElement);
+			grid.Children.Add(ButtonContainer);
+
+			popup.XamlRoot = mediaPlayerElement.XamlRoot;
+			popup.HorizontalOffset = 0;
+			popup.VerticalOffset = 0;
+			popup.ShouldConstrainToRootBounds = false;
+			popup.VerticalAlignment = Microsoft.UI.Xaml.VerticalAlignment.Center;
+			popup.HorizontalAlignment = Microsoft.UI.Xaml.HorizontalAlignment.Center;
+			popup.Child = grid;
+
+			if (!popup.IsOpen)
+			{
+				popup.IsOpen = true;
+			}
+		}
+	}
+
+	static AppWindow GetAppWindowForCurrentWindow()
+	{
+		// let's cache the CurrentPage here, since the user can navigate or background the app
+		// while this method is running
+		var currentPage = CurrentPage;
+
+		if (currentPage?.GetParentWindow().Handler.PlatformView is not MauiWinUIWindow window)
+		{
+			throw new InvalidOperationException();
```

Comment:
can we add helpful message for the user?

Reply:
```throw new InvalidOperationException($"{nameof(Application.Current.MainPage)} cannot be null.");```